modules/core/03-connection/keeper/handshake.go

```diff
@@ -223,28 +179,19 @@ func (k Keeper) ConnOpenAck(
 		return sdkerrors.Wrap(types.ErrConnectionNotFound, connectionID)
 	}
 
-	// Verify the provided version against the previously set connection state
-	switch {
-	// connection on ChainA must be in INIT or TRYOPEN
-	case connection.State != types.INIT && connection.State != types.TRYOPEN:
-		return sdkerrors.Wrapf(
-			types.ErrInvalidConnectionState,
-			"connection state is not INIT or TRYOPEN (got %s)", connection.State.String(),
-		)
-
-	// if the connection is INIT then the provided version must be supproted
-	case connection.State == types.INIT && !types.IsSupportedVersion(version):
+	// verify the previously set connection state
+	if connection.State != types.INIT {
 		return sdkerrors.Wrapf(
 			types.ErrInvalidConnectionState,
-			"connection state is in INIT but the provided version is not supported %s", version,
+			"connection state is not INIT (got %s)", connection.State.String(),
 		)
+	}
 
-	// if the connection is in TRYOPEN then the version must be the only set version in the
-	// retreived connection state.
-	case connection.State == types.TRYOPEN && (len(connection.Versions) != 1 || !proto.Equal(connection.Versions[0], version)):
+	// ensure selected version is supported
+	if !types.IsSupportedVersion(types.ProtoVersionsToExported(connection.Versions), version) {
```

Comment:
I think it should be possible to change the API's to use the proto version type (remove the switch between exported/proto). We would remove the `GetVersions` function from the connection interface which is fine since we only have one connection type (we can cast to connectiontypes.ConnectionEnd) which is already what occurs

Reply:
opened #1689 and #1690